Source/WebKit/UIProcess/WebContextMenuListenerProxy.h

```diff
@@ -31,21 +31,16 @@
 #include <wtf/Forward.h>
 #include <wtf/WeakPtr.h>
 
-namespace WebKit {
-struct WebContextMenuListenerProxyClient;
-}
-
-namespace WTF {
-template<typename T> struct IsDeprecatedWeakRefSmartPointerException;
-template<> struct IsDeprecatedWeakRefSmartPointerException<WebKit::WebContextMenuListenerProxyClient> : std::true_type { };
-}
-
 namespace WebKit {
 
 class WebContextMenuItem;
 
 struct WebContextMenuListenerProxyClient : CanMakeWeakPtr<WebContextMenuListenerProxyClient> {
     virtual ~WebContextMenuListenerProxyClient() = default;
+
+    virtual void ref() const = 0;
+    virtual void deref() const = 0;
```

Comment:
I believe we have consensus now: Let's inherit AbstractRefCounted here instead of declaring these manually.

Reply:
Indeed, please subclass `AbstractRefCounted`.